### PR TITLE
feat: configurable provisoner log retention

### PR DESCRIPTION
**Pull request by ahammond**

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -428,6 +444,8 @@ public readonly retention: Duration;
 - *Type:* aws-cdk-lib.Duration
 - *Default:* Duration.days(1) This should pass through, but nope. So, we're duplicating the default.
 
+RDS backup retention.
```
Comment (all-mike) [inline API.md:447]: hm can you explain why this RDS line got added?

**Review by all-mike — APPROVED** [review]
Looks good!

File: src/aurora.ts [hunk 2]
```diff
@@ -299,7 +308,7 @@ export class Aurora extends Construct {
           bundling: { minify: true },
           entry: join(__dirname, 'aurora.activity-stream.ts'),
           handler,
-          logRetention: aws_logs.RetentionDays.ONE_WEEK,
+          logRetention: props.lambdaLogRetention ?? aws_logs.RetentionDays.TWO_YEARS,
```
Comment (nathanhruby) [inline src/aurora.ts:311]: How about 90 days?

**Review by all-mike — APPROVED** [review]
Looks good!

**Review by all-mike — APPROVED** [review]
Looks good!

File: API.md (hunk 1 above)
Comment (ahammond) [reply API.md:447]: To differentiate between the various flavours of data retention. It's a pretty trivial change, but yeah, should probably be in a separate PR.

**Pull request merged**